src/Native/System.Native/pal_console.cpp

```diff
@@ -35,9 +35,10 @@ extern "C" int32_t GetWindowSize(WinSize* windowSize)
 #endif
 }
 
-extern "C" int32_t IsATty(int fd)
+extern "C" int32_t IsATty(intptr_t fd)
 {
-    return isatty(fd);
+    assert(INT32_MIN <= fd && fd <= INT32_MAX);
```

Comment:
My understanding was that these asserts are to help us catch bugs in our code, and aren't really to validate a "public" contract.

We've treated the shim APIs as internal in other places:  https://github.com/dotnet/corefx/pull/4267#discussion_r43602869